From 8651d8953707217c82fb01f657868ab9a4b7c51c Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Thu, 20 Nov 2025 17:01:28 +0100
Subject: [PATCH] init: Add option to install systemd services without systemd
 installed

Allows the sub-packaging of the service in Alpine, where systemd is not
available but packaging of systemd services is allowed (e.g. for postmarketOS)
---
 CMakeLists.txt      | 1 +
 init/CMakeLists.txt | 8 ++++++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 861717b..c97cc10 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,7 @@ pkg_check_modules(ACCOUNTSSERVICE REQUIRED accountsservice)
 pkg_get_variable(AS_INTERFACES_DIR accountsservice interfacesdir)
 
 option(ENABLE_LIBHYBRIS "Enable libhybris support" ON)
+option(ENFORCE_SYSTEMD_SERVICE "Enable installation of systemd services" OFF)
 
 if (ENABLE_LIBHYBRIS)
   pkg_check_modules(ANDROID_HEADERS android-headers)
diff --git a/init/CMakeLists.txt b/init/CMakeLists.txt
index e27a009..b19e676 100644
--- a/init/CMakeLists.txt
+++ b/init/CMakeLists.txt
@@ -3,9 +3,13 @@
 ##
 
 pkg_check_modules(SYSTEMD systemd)
-if (${SYSTEMD_FOUND})
+if (SYSTEMD_FOUND OR ENFORCE_SYSTEMD_SERVICE)
 
-    pkg_get_variable(SYSTEMD_SYSTEM_DIR systemd systemdsystemunitdir)
+    if (ENFORCE_SYSTEMD_SERVICE)
+        set (SYSTEMD_SYSTEM_DIR "/usr/lib/systemd/system/")
+    elseif(${SYSTEMD_FOUND})
+        pkg_get_variable(SYSTEMD_SYSTEM_DIR systemd systemdsystemunitdir)
+    endif()
     message (STATUS "${SYSTEMD_SYSTEM_DIR} is the systemd system unit file install dir")
 
     set (SYSTEMD_SYSTEM_NAME "${CMAKE_PROJECT_NAME}.service")
-- 
GitLab

